src/pow.cpp

```diff
@@ -61,7 +61,19 @@ unsigned int CalculateNextWorkRequired(const CBlockIndex* pindexLast, int64_t nF
     // Retarget
     const arith_uint256 bnPowLimit = UintToArith256(params.powLimit);
     arith_uint256 bnNew;
-    bnNew.SetCompact(pindexLast->nBits);
+
+    // Special difficulty rule for Testnet4
+    if (params.hashGenesisBlock == uint256S("0x00000000da84f2bafbbc53dee25a72ae507ff4914b867c565be350b0da8bf043")) {
```

Comment:
I think I have seen someone mention something along the lines before, but it would be nice to have a global constant for the testnet 4 Genesis Block Hash instead of it being hard-coded like this.

Reply:
I had planned to do this as a follow-up but I added a new property `enforce_BIP94` to handle this now.